psdisk/psdisk.c

```diff
@@ -117,199 +116,220 @@ static void psdisk_showPartsTable(void)
 
 	fseek(psdisk_common.file, 0L, SEEK_END);
 	printf("\n");
-	printf(BOLDWHITE "File system %s: %ld bytes\n"RESET, psdisk_common.fileName, ftell(psdisk_common.file));
-	printf("Memory size %u bytes\n", psdisk_common.mem.memSize);
-	printf("Units: sector %u bytes \n", psdisk_common.mem.sectorSize);
+	printf(BOLDWHITE "Partition table %s: %ld bytes\n"RESET, psdisk_common.fileName, ftell(psdisk_common.file));
+	printf("Memory size: %u bytes\n", psdisk_common.memsz);
+	printf("Block size: %u bytes\n", psdisk_common.blksz);
 	printf("\n");
-	printf(BOLDWHITE PARTITION_HEADER_ALIGMENT RESET, "Name", "Start", "End", "Sectors", "Size", "Type");
+	printf(BOLDWHITE PARTITION_HEADER_ALIGMENT RESET, "Name", "Start", "End", "Blocks", "Size", "Type");
 	LIST_FOREACH(node, &psdisk_common.list, ptrs) {
-		printf(PARTITION_DATA_ALIGMENT, node->pHeader.name, node->pHeader.offset, node->pHeader.offset + node->pHeader.size,
-			   node->pHeader.size / psdisk_common.mem.sectorSize, node->pHeader.size, psdisk_getTypeName(node->pHeader.type));
+		printf(PARTITION_DATA_ALIGMENT, node->part.name, node->part.offset, node->part.offset + node->part.size,
+			   node->part.size / psdisk_common.blksz, node->part.size, psdisk_getTypeName(node->part.type));
 	}
 	printf("\n");
 }
 
 
-/* Callbacks to ptable library */
+/* Partition table read/write functions */
 
-static ssize_t psdisk_readImg(unsigned int addr, void *buff, size_t bufflen)
+static ptable_t *psdisk_readImg(void)
 {
-	int offs;
+	ptable_t *ptable;
+	uint32_t count, size;
 
-	if (PTABLE_WITH_OFFSET(psdisk_common.opts))
-		offs = addr;
-	else
-		offs = addr - (psdisk_common.mem.memSize - psdisk_common.mem.sectorSize);
+	if (fseek(psdisk_common.file, 0, SEEK_SET) != 0) {
+		return NULL;
+	}
 
-	if (fseek(psdisk_common.file, offs, SEEK_SET) != 0) {
-		fprintf(stderr,"Cannot change file position, err: %s.\n", strerror(errno));
-		return -1;
+	if (fread(&count, 1, sizeof(count), psdisk_common.file) != sizeof(count)) {
```

Comment:
Nitpick - I think intended use is along these lines
```suggestion
	if (fread(&count, sizeof(count), 1, psdisk_common.file) != 1) {
```

Reply:
If you wish to change it, there're more occurrences. I doesn't really matter, though

Reply:
I guess it's better to use `fread()`\`fwrite()` in the intended way :P Fixed in the next force-push.